apps/web/hooks/useDrawChart.tsx

```diff
@@ -15,28 +15,38 @@ const useDrawChart = (options: EChartOption, isLoading?: boolean) => {
   const chartRef = useRef<HTMLDivElement>(null);
   const [chart, setChart] = useState<echarts.ECharts | null>(null);
 
+  const resizeObserverRef = useRef<ResizeObserver | null>(null);
+
+  //Creating a chart once at initial render
   useEffect(() => {
+    if (!chartRef.current) return;
+
     const chartInstance = echarts.init(chartRef.current);
-    chartInstance.setOption(options);
     setChart(chartInstance);
 
-    const resizeObserver = new window.ResizeObserver((entries) => {
-      entries.map(({ target }) => {
+    resizeObserverRef.current = new window.ResizeObserver((entries) => {
+      entries.forEach(({ target }) => {
         if (target instanceof HTMLDivElement) {
           const instance = echarts.getInstanceByDom(target);
-          if (instance) {
-            instance.resize();
-          }
+          instance?.resize();
         }
       });
     });
-    resizeObserver.observe(chartRef.current!);
+
+    resizeObserverRef.current.observe(chartRef.current);
 
     return () => {
-      resizeObserver.disconnect();
+      resizeObserverRef.current?.disconnect();
       chartInstance.dispose();
+      setChart(null);
```

Comment:
Avoid using setState in unmount function since at this moment your component is unmounted and there is no state